products/data_warehouse/backend/hogql_fixer_ai.py

```diff
@@ -86,24 +123,42 @@ def get_lexer_contents() -> str:
 
 - Don't update any other part of the query if it's not relevant to the error, including rewriting shorthand clickhouse SQL to the full version.
 - Don't change the capitalisation of the query if it's not relevant to the error, such as rewriting `select` as `SELECT` or `from` as `FROM`
+- Don't convert syntax to an alternative if it's not relevant to the error, such as changing `toIntervalDay(1)` to `INTERVAL 1 DAY`
 - There may also be more than one error in the syntax.
 
+{{schema_description}}
+
 Below is the current HogQL query and the error message
 """
 
 
-class HogQLQueryFixerTool(MaxTool):
-    name: str = "fix_hogql_query"
-    description: str = "Fixes any error in the current HogQL query"
-    thinking_message: str = "Fixing errors in the SQL query"
-    root_system_prompt_template: str = SQL_ASSISTANT_ROOT_SYSTEM_PROMPT
+def _get_schema_description(ai_context: dict[Any, Any], hogql_context: HogQLContext, database: Database) -> str:
+    serialized_database = serialize_database(hogql_context)
 
-    def _run_impl(self) -> tuple[str, str | None]:
-        database = create_hogql_database(self._team_id)
-        hogql_context = HogQLContext(team_id=self._team_id, enable_select_queries=True, database=database)
+    try:
+        query = ai_context.get("hogql_query", None)
+        if not query:
+            # Dummy exception to get into the except block
+            raise Exception()
+
+        select = parse_select(query)
+        tables_in_query = get_table_names(select)
+        table_fields_str = ""
 
-        serialized_database = serialize_database(hogql_context)
-        schema_description = "\n\n".join(
+        for table_name in tables_in_query:
+            serialized_table = serialized_database.get(table_name, None)
+            if serialized_table:
+                table_fields_str = table_fields_str + f"Table `{table_name}` with fields:\n"
+                table_fields_str = table_fields_str + "\n".join(
+                    f"- {field.name} ({field.type})" for field in serialized_table.fields.values()
+                )
+
+        if len(table_fields_str) != 0:
+            schema_description = (
+                "This is the schema of tables currently used in the provided query:\n\n" + table_fields_str
+            )
+    except:
```

Comment:
guess this is ok for now but might want to improve the error handling in future just so we're not masking any genuine errors